backend.go

```diff
@@ -15,7 +15,10 @@ import (
 	"unsafe"
 )
 
-var currentBackend Backend
+var currentBackend backendCExpose
+
+// TODO: Maybe we should get rid of it?
+var textureManager TextureManager
```

Comment:
@AllenDang I'm having trouble here...
My solution is kinda stupid but can't find anything better.
Our texture stuff from `texture.go` needs access to backend in order to be loaded... And I don't want to make it debendent on BackendFlagsT :confused:
Maybe we should remove whole texture.go and let user implement it themselves?

Reply:
I prefer not to remove texture.go... like the default backend which could be removed by build via flag, maybe we could treat texture stuff as a bundle with backend, so to implement a backend, three things should be implemented, platform, render and texture.

Reply:
ok, so lets leave it as-is for now. Gonna merge this